src/bionics.cpp

```diff
@@ -860,7 +886,33 @@ void player::process_bionic( int b )
             }
         }
     }
+    if( !bio.info().fuel_opts.empty() && !bio.is_muscle_powered() ) {
+        for( const itype_id fuel : get_fuel_available( bio.id ) ) {
+            const item tmp_fuel( fuel.c_str() );
+            int temp = std::stoi( get_value( fuel ) );
+            if( power_level + tmp_fuel.fuel_energy() * ( bio.info().fuel_efficiency / 100 ) >
+                max_power_level ) {
+                add_msg_player_or_npc( m_info, _( "Your %s turns off to not waste fuel." ),
+                                       _( "<npcname>'s %s turns off to not waste fuel." ), bio.info().name );
+                bio.powered = false;
+                deactivate_bionic( b, true );
 
+            } else {
+                if( temp > 0 ) {
+                    temp -= 1;
+                    charge_power( tmp_fuel.fuel_energy() * ( bio.info().fuel_efficiency / 100 ) );
+                    set_value( fuel, std::to_string( temp ) );
+                    update_fuel_storage( item( fuel ) );
```

Comment:
Looks like this is repeated from the code above. Should be in a separate function instead.

Reply:
moved to burn_fuel()